osbenchmark/builder/utils/jdk_resolver.py

```diff
@@ -0,0 +1,103 @@
+import os
+import re
+
+from osbenchmark.exceptions import SystemSetupError
+from osbenchmark.utils import io
+
+
+class JdkResolver:
+    def __init__(self, executor):
+        self.executor = executor
+
+    def resolve_jdk_path(self, host, majors):
+        """
+        Resolves the path to the JDK with the provided major version(s). It checks the versions in the same order specified in ``majors``
+        and will return the first match. To achieve this, it first checks the major version x in the environment variable ``JAVAx_HOME``
+        and falls back to ``JAVA_HOME``. It also ensures that the environment variable points to the right JDK version.
+
+        If no appropriate version is found, a ``SystemSetupError`` is raised.
+
+        :param host: The host on which to resolve the JDK path
+        :param majors: Either a list of major versions to check or a single version as an ``int``.
+        :return: A tuple of (major version, path to Java home directory).
+        """
+        if isinstance(majors, int):
```

Comment:
Nit: Are we following pep8? If so, this method should be updated to `is_instance`

Reply:
Not sure tbh, I've been relying on `make precommit` for code styling. Should we be following pep8?

Reply:
Update: spoke offline, I misunderstood that isinstance() is a built-in function and not in reference to our cluster and AOS terminology.

Reply:
This is a python built in function. We can't do anything about the formatting :)

https://docs.python.org/3/library/functions.html#isinstance